src/elfpy/pricing_models/yieldspace.py

```diff
@@ -56,42 +49,49 @@ def calc_lp_out_given_tokens_in(
 
         """
         assert d_base > 0, f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected d_base > 0, not {d_base}!"
-        assert (
-            share_reserves >= 0
-        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected share_reserves >= 0, not {share_reserves}!"
-        assert (
-            bond_reserves >= 0
-        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected bond_reserves >= 0, not {bond_reserves}!"
-        assert (
-            base_buffer >= 0
-        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected base_buffer >= 0, not {base_buffer}!"
-        assert (
-            lp_reserves >= 0
-        ), f"pricing_models.calc_lp_out_given_tokens_in: ERROR: expected lp_reserves >= 0, not {lp_reserves}!"
+        assert market_state.share_reserves >= 0, (
+            "pricing_models.calc_lp_out_given_tokens_in: ERROR:  "
+            f"Expected share_reserves >= 0, not {market_state.share_reserves}!"
+        )
+        assert market_state.bond_reserves >= 0, (
+            "pricing_models.calc_lp_out_given_tokens_in: ERROR: "
+            f"Expected bond_reserves >= 0, not {market_state.bond_reserves}!"
+        )
+        assert market_state.base_buffer >= 0, (
+            "pricing_models.calc_lp_out_given_tokens_in: ERROR: "
+            f"Expected base_buffer >= 0, not {market_state.base_buffer}!"
+        )
+        assert market_state.lp_reserves >= 0, (
+            "pricing_models.calc_lp_out_given_tokens_in: ERROR: "
+            f"Expected lp_reserves >= 0, not {market_state.lp_reserves}!"
+        )
```

Comment:
I think we'll ultimately want to de-couple assertions from these LP functions in the same way that we did with the trading functions assertions. This strengthens the argument that LP and trading logic should be separately IMO since it would make the naming for these assertion functions cleaner.